### PR TITLE
Use vue

### PR DESCRIPTION
**Pull request by saifxhatem**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,10 @@
 /node_modules
 /random_snippets
 /resources/views/welcome.blade.phptmp
+/resources/views/welcome.blade.phpbk
```
Comment (Mhany95) [inline .gitignore:4]: Remove them from your gitignore

File: resources/js/components/styles/display-addresses-card.css [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.card {
+  font-family: "Times New Roman", Times, serif;
+}
+
+li {
```
Comment (Mhany95) [inline resources/js/components/styles/display-addresses-card.css:5]: use a class

File: resources/js/components/AlertButtonComponent.vue [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  props: {
+    address: {
+      type: String,
+      default: "My Address"
+    }
+  },
+  methods: {
+    alert: function (event) {
+      alert('You chose address: ' + this.address)
```
Comment (Mhany95) [inline resources/js/components/AlertButtonComponent.vue:20]: 1- Don't use '+' for string concatenation 
2- Add a guard condition for "address"

File: resources/js/app.js [hunk 4]
```diff
@@ -1 +1,37 @@
...
+
+
+window.Vue = require('vue');
+import VueRouter from 'vue-router'
+import DisplayAddressByID from './components/DisplayAddressByID.vue'
+import DisplayAddresses from './components/DisplayAddresses.vue'
+
+Vue.use(VueRouter)
+
+const Foo = { template: '<div>foo</div>' }
```
Comment (Mhany95) [inline resources/js/app.js:11]: remove dummy data (Foo and Bar)

File: resources/js/app.js [hunk 5]
```diff
@@ -1 +1,37 @@
...
+const Bar = { template: '<div>bar</div>' }
+
+
+const routes = [
+  { name : 'foo', path: '/foo', component: Foo },
+  { name: 'bar', path: '/bar', component: Bar },
+  { name: 'displayAddressByID', path: '/fetchaddr', component: DisplayAddressByID, props: true},
```
Comment (Mhany95) [inline resources/js/app.js:18]: follow a unified casing with path names

File: resources/js/components/ButtonComponent.vue [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+</template>
+
+
+<script>
+
+export default {
+
+  props: {
+    alert_text: {
```
Comment (Mhany95) [inline resources/js/components/ButtonComponent.vue:13]: not used

File: resources/js/components/DisplayAddressByID.vue [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+          addresses: [],
+          title: "Address Info:"
+        }
+      },
+
+      mounted() {
+        this.loadAddresses();
+      },
+
+      created() {
```
Comment (Mhany95) [inline resources/js/components/DisplayAddressByID.vue:44]: remove empty method

File: resources/js/components/DisplayAddresses.vue [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+              this.addresses = response.data;
+            else
+              console.log("Retrieved bad data.")
+
+          })
+          .catch(function (error) {
+            console.log(error);
+          });
+        },
+        alertAddress: function(address) {
```
Comment (Mhany95) [inline resources/js/components/DisplayAddresses.vue:51]: remove unused function

File: resources/js/components/DisplayAddresses.vue [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+          let base_message = "You chose this address: ";
+          let full_message = base_message.concat(address);
+          alert(full_message);
+        },
+
+        redirectToAddress: function(address_id)
+        {
+          let base_url = "/fetch-address-by-id/";
```
Comment (Mhany95) [inline resources/js/components/DisplayAddresses.vue:59]: use router

File: resources/js/components/DisplayUsers.vue [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+        this.loadUsers();
+      },
+      methods: {
+        loadUsers: function () {
+          //get Request
+          //assign results
+          //catch $errors
+          axios.get('/list-users')
+          .then((response) => {
+            console.log(response.data);
```
Comment (Mhany95) [inline resources/js/components/DisplayUsers.vue:33]: remove console log

File: resources/js/components/InsertAddressFormComponent.vue [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+      return {
+        formData: {
+          user_id: null,
+          address: null
+        }
+      }
+    },
+    
+    mounted() {
+        console.log('Component mounted.')
```
Comment (Mhany95) [inline resources/js/components/InsertAddressFormComponent.vue:59]: remove console log

File: resources/js/components/InsertAddressFormComponent.vue [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+    },
+
+    methods:
+    {
+      postData(e)
+      {
+          console.log(this.formData);
```
Comment (Mhany95) [inline resources/js/components/InsertAddressFormComponent.vue:66]: remove console log

File: resources/views/display_addresses.blade.php [hunk 13]
```diff
@@ -23,19 +23,11 @@
...
-        <tr>
-        <td>{{ $address->address }}<br></td>
-
-        </tr>
-      @endforeach
-    </div>
-
-    </div>
+  <div id= "app">
+    <router-view></router-view>
```
Comment (Mhany95) [inline resources/views/display_addresses.blade.php:27]: ?

File: resources/views/display_addresses_by_id.blade.php [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+            <ul>
+               @foreach ($errors->all() as $error)
+                  <li>{{ $error }}</li>
+               @endforeach
+            </ul>
+         </div>
+      @endif
+
+  <div id= "app">
+    <router-link to="/foo">Go to Foo</router-link>
```
Comment (Mhany95) [inline resources/views/display_addresses_by_id.blade.php:27]: ?

Comment (saifxhatem) [reply resources/views/display_addresses_by_id.blade.php:27]: This was the test page and should not have been committed

**Pull request merged**